resources/aws_ec2.go

```diff
@@ -829,9 +836,29 @@ func (obj *AwsEc2Res) snsPostHandler(w http.ResponseWriter, req *http.Request) {
 		http.Error(w, "Invalid request method", http.StatusMethodNotAllowed)
 		return
 	}
-	post, _ := ioutil.ReadAll(req.Body)
-	if obj.debug {
-		log.Printf("%s: Post: %s", obj, string(post))
+	// decode json
+	decoder := json.NewDecoder(req.Body)
+	var post postData
+	err := decoder.Decode(&post)
+	if err != nil {
+		select {
+		case obj.awsChan <- &chanStruct{
+			err: errwrap.Wrapf(err, "error decoding post"),
+		}:
+		case <-obj.closeChan:
+		}
+		return
+	}
+	if post.Type == "SubscriptionConfirmation" {
```

Comment:
This feels like an event saying "we're now ready" to watch events or something...
If my understanding is correct, that means this should be the obj.Running() runner, no?
Probably best to understand this a bit better, and maybe add a comment.

Reply:
Yeah. I have another commit going which adds a run field to chanStruct and uses it to signal snsWatch, which then runs obj.Running(). It might be simpler to just run it here so I don't need to modify chanStruct. What do you prefer?